### PR TITLE
[FCOS] support libvirt install

### PR DESCRIPTION
**Pull request by vrutkovs**
* Update links to FCOS qcow
* Extract xz-compressed images in libvirt

Requires https://github.com/openshift/release/pull/5933 to add libvirt test.

Fixes https://github.com/openshift/okd/issues/15

### CODE REVIEW

File: pkg/tfvars/internal/cache/cache.go [hunk 1]
```diff
@@ -246,6 +248,34 @@ func decompressFileGzip(src, dest string) error {
...
+		return err
+	}
+
+	writer, err := os.Create(dest)
+	// defer os.Remove(dest)
+	defer writer.Close()
+	if err != nil {
+		return err
+	}
+	defer writer.Close()
```
Comment (cfergeau) [inline pkg/tfvars/internal/cache/cache.go]: there's an extra `defer` here.

File: pkg/tfvars/internal/cache/cache.go [hunk 2]
```diff
@@ -246,6 +248,34 @@ func decompressFileGzip(src, dest string) error {
...
+
+	return nil
+}
+
+func decompressFileXZ(src, dest string) error {
+	logrus.Debugf("decompressFileXZ %s %s", src, dest)
+	xzfile, err := os.Open(src)
+	if err != nil {
+		return err
+	}
```
Comment (cfergeau) [inline pkg/tfvars/internal/cache/cache.go]: no defer `xzfile.Close()`?

File: pkg/tfvars/internal/cache/cache.go (hunk 1 above)
Comment (vrutkovs) [reply pkg/tfvars/internal/cache/cache.go]: Fixed

File: pkg/tfvars/internal/cache/cache.go (hunk 2 above)
Comment (vrutkovs) [reply pkg/tfvars/internal/cache/cache.go]: Fixed

**Comment by vrutkovs** [comment]
/hold

until https://github.com/openshift/release/pull/5933 lands so that it could be properly tested

File: data/data/rhcos.json [hunk 3]
```diff
@@ -111,11 +111,11 @@
             "size": 647639040
         },
         "qemu": {
-            "path": "rhcos-43.80.20191002.1-qemu.x86_64.qcow2",
-            "sha256": "29e867427e84b64970b8e0408b135b13f6e315b82882be5601e19529d4463773",
-            "size": 700315704,
-            "uncompressed-sha256": "b4cd810376dca356f987e2e7f8609f84d63720d5ad19d3101f84e82d1918c85e",
-            "uncompressed-size": 1889337344
+            "path": "fedora-coreos-30.20191014.0-qemu.x86_64.qcow2.xz",
```
Comment (stbenjam) [inline data/data/rhcos.json]: This is going to break baremetal, we rely on the QEMU image for the bootstrap node - it seems odd to have mixed RHCOS/FCOS content here, shouldn't it be one or the other for everything?

Comment (stbenjam) [reply data/data/rhcos.json]: Also the next time someone runs the hack script to update the RHCOS build this will all get overwritten.

Comment (vrutkovs) [reply data/data/rhcos.json]: Baremetal IPI is currently not supported on OKD-on-FCOS - Ironic bits need to be replaced with UBI8 + RDO, hack script updated etc. I'll make another PR once it gets there

Comment (stbenjam) [reply data/data/rhcos.json]: Right I get that it won't work with OKD-on-FCOS, but this is breaking baremetal IPI entirely. Is all of openshift/install master moving to FCOS and too bad for any platforms that don't work with it yet?

Comment (stbenjam) [reply data/data/rhcos.json]: My concern here is baremetal IPI has been broken for weeks, and we're very close to recovering from it and having working CI again. Now, with this change, our Metal3 CI can't be run against anything in master, so we really have no idea if some future commits break us until it ends up in a release. :-(

Comment (stbenjam) [reply data/data/rhcos.json]: Sorry for the noise, it was just pointed out to me this isn't going to the master branch

File: pkg/rhcos/builds.go [hunk 4]
```diff
@@ -20,12 +20,14 @@ type metadata struct {
 	BaseURI string `json:"baseURI"`
 	Images  struct {
 		QEMU struct {
-			Path   string `json:"path"`
-			SHA256 string `json:"sha256"`
+			Path               string `json:"path"`
+			SHA256             string `json:"sha256"`
+			UncompressedSHA256 string `json:"uncompressed-sha256"`
```
Comment (zeenix) [inline pkg/rhcos/builds.go]: Some info in the commit log details on why this change is needed, would be good.

File: data/data/rhcos.json (hunk 3 above)
Comment (zeenix) [reply data/data/rhcos.json]: > it seems odd to have mixed RHCOS/FCOS content here, shouldn't it be one or the other for everything?

@vrutkovs I think this very important question from @stbenjam wasn't answered. If the answer is that we're just moving everything to FCOS, I think we should rename the file and (more importantly) commit log should reflect the reason for the change.

Comment (cgwalters) [reply data/data/rhcos.json]: This PR is only for the `fcos` branch, it won't affect main development on `master`.

**Comment by vrutkovs** [comment]
>@vrutkovs I think this very important question from @stbenjam wasn't answered. If the answer is that we're just moving everything to FCOS, I think we should rename the file and (more importantly) commit log should reflect the reason for the change.

Both changes would be addressed in followup PRs. Baremetal supports depends on openstack images rebuilt from RDO (instead of OSP). `rhcos.json` needs a consensus  from architects on the chosen name. At this point we keep the same name to avoid extra changes during regular rebases from master

**Comment by vrutkovs** [comment]
No longer required - all fixes have been merged in the branch
